pkg/healthcheck/healthcheck.go

```diff
@@ -1514,16 +1523,20 @@ func (hc *HealthChecker) runCheck(categoryID CategoryID, c *Checker, observer Ch
 			return true
 		}
 
+		hintBaseURL := category.hintBaseURL
```

Comment:
I think it would be simpler if we set the defaultHintBaseURL into the Category when it's created by `NewCategory` so that we don't need any defaulting logic here.